### PR TITLE
Integrated Redis

### PR DESCRIPTION
**Pull request by matttychen**

### CODE REVIEW

File: web/src/view/map/Room.tsx [hunk 1]
```diff
@@ -79,7 +80,8 @@ export function Room ({ event, user }: RoomProps) {
   React.useEffect(() => {
     switchTable(getApolloClient(), {
       eventTableId: mainEventTableId,
-      participantId: user.user.id
+      participantId: user.user.id,
+      participantName: user.user.name
```
Comment (pacbac) [inline web/src/view/map/Room.tsx:84]: should not need to pass user name, since it can be retrieved by user id. We want minimize our payload. Even though ik this might save a search on the server side, we should not rely on the client to supply anything more than essential information, since username could've been manipulated on the client side to be inconsistent from user id.

File: web/src/view/map/Square.tsx [hunk 2]
```diff
@@ -73,7 +73,8 @@ export function Square ({
 
     switchTable(getApolloClient(), {
       eventTableId,
-      participantId: user.user.id
+      participantId: user.user.id,
+      participantName: user.user.name
```
Comment (pacbac) [inline web/src/view/map/Square.tsx:77]: same as other comment

File: web/src/view/page/EventCreateForm.tsx [hunk 3]
```diff
@@ -46,6 +46,7 @@ export function EventCreateForm(props: EventCreateForm) {
       toast("Cannot create: Form input must be valid. All mandatory fields must be filled with positive/valid values.")
       return
     }
+    console.log(startTime, new Date())
```
Comment (pacbac) [inline web/src/view/page/EventCreateForm.tsx:50]: remove console log

File: web/src/view/map/Room.tsx [hunk 4]
```diff
@@ -56,7 +56,8 @@ export function Room ({ event, user }: RoomProps) {
   const leaveTableOnUnmount = () => {
     switchTable(getApolloClient(), {
       eventTableId: null, // leave event
-      participantId: user.user.id
+      participantId: user.user.id,
+      participantName: user.user.name
```
Comment (pacbac) [inline web/src/view/map/Room.tsx:60]: same here as well

File: server/src/graphql/schema.graphql [hunk 5]
```diff
@@ -17,16 +17,19 @@ schema {
 type Query {
   # Returns the logged-in user making the query, or null if anonymous.
   self: User
-  usersAtTable(tableId: Int!): [User!]!
+  usersAtTable(tableId: Int!): [TableParticipant!]!
```
Comment (pacbac) [inline server/src/graphql/schema.graphql:20]: why was this changed? `TableParticipants` are users, better not to prevent multiple data types that accomplish the same thing, and `User` is more generic and flexible

File: web/src/view/map/Room.tsx (hunk 1 above)
Comment (matttychen) [reply web/src/view/map/Room.tsx:84]: got it

File: server/src/graphql/schema.graphql (hunk 5 above)
Comment (matttychen) [reply server/src/graphql/schema.graphql:20]: fixed

**Comment by matttychen** [comment]
don't merge yet
